packages/zoe/src/contracts/coveredCall.js

```diff
@@ -18,40 +18,53 @@ import harden from '@agoric/harden';
 
 import { makeHelpers } from './helpers/userFlow';
 
-export const makeContract = harden((zoe, terms) => {
-  const { rejectOffer, hasValidPayoutRules, swap } = makeHelpers(
+export const makeContract = harden((zoe, _terms) => {
+  const { roleNames, issuers } = zoe.getInstanceRecord();
+  const { swap, assertRoleNames, rejectIfNotOfferRules } = makeHelpers(
     zoe,
-    terms.issuers,
+    roleNames,
+    issuers,
   );
-  const ASSET_INDEX = 0;
-  const PRICE_INDEX = 1;
+
+  assertRoleNames(roleNames, harden(['UnderlyingAsset', 'StrikePrice']));
 
   const makeCallOptionInvite = sellerHandle => {
     const seat = harden({
-      exercise: () =>
-        swap(sellerHandle, inviteHandle, `The covered call option is expired.`),
+      exercise: () => {
+        const expected = harden({
+          offer: ['StrikePrice'],
+          want: ['UnderlyingAsset'],
+        });
+        rejectIfNotOfferRules(inviteHandle, expected);
+        return swap(
+          sellerHandle,
+          inviteHandle,
+          `The covered call option is expired.`,
+        );
+      },
     });
-    const { payoutRules, exitRule } = zoe.getOffer(sellerHandle);
+
+    const { userOfferRules: sellerOfferRules } = zoe.getOffer(sellerHandle);
+
     const { invite: callOption, inviteHandle } = zoe.makeInvite(seat, {
       seatDesc: 'exerciseOption',
-      expirationDate: exitRule.deadline,
-      timerAuthority: exitRule.timer,
-      underlyingAsset: payoutRules[ASSET_INDEX].amount,
-      strikePrice: payoutRules[PRICE_INDEX].amount,
+      expirationDate: sellerOfferRules.exit.AfterDeadline.deadline,
```

Comment:
Stylistic suggestion: it's hard to see the important differentiating part of these.  Can you add anothe rlevel of destructuring so each line looks more like: `expirationDate: sellerExit.AfterDeadline.deadline`?

Reply:
Sure thing. Added a layer of destructuring